legend-engine-xt-nonrelationalStore-mongodb-grammar/src/main/java/org/finos/legend/engine/language/mongodb/schema/grammar/from/MongoDBQueryDeserializer.java

```diff
@@ -0,0 +1,603 @@
+                typeValue.value = jsonNode.longValue();
+                literalValue.value = typeValue;
+            }
+            else if (jsonNode.isDouble() || jsonNode.isFloat() || jsonNode.isBigDecimal())
+            {
+                DecimalTypeValue typeValue = new DecimalTypeValue();
+                typeValue.value = new BigDecimal(jsonNode.asText());
+                literalValue.value = typeValue;
+            }
+            else if (jsonNode.isNull())
+            {
+                literalValue.value = new NullTypeValue();
+            }
+        }
+        else if (jsonNode.isObject())
+        {
+            ObjectTypeValue objTypeValue = new ObjectTypeValue();
+            List<KeyValuePair> kvPairs = getKeyValuePairs(jsonNode);
+            objTypeValue.keyValues = kvPairs;
+            literalValue.value = objTypeValue;
+        }
+        else if (jsonNode.isArray())
+        {
+            ArrayTypeValue arrayTypeValue = new ArrayTypeValue();
+            List<BaseTypeValue> items = Lists.mutable.of();
+            for (JsonNode item : jsonNode)
+            {
+                BaseTypeValue itemValue = getLiteralValueFromNode(item).value;
+                items.add(itemValue);
+            }
+            arrayTypeValue.items = items;
+            literalValue.value = arrayTypeValue;
+        }
+        return literalValue;
+
+
+    }
+
+    private static List<KeyValuePair> getKeyValuePairs(JsonNode jsonNode)
+    {
+        Iterator<Map.Entry<String, JsonNode>> literalObjFields = jsonNode.fields();
+        List<KeyValuePair> kvPairs = Lists.mutable.empty();
+        while (literalObjFields.hasNext())
+        {
+            Map.Entry<String, JsonNode> literalObjEntry = literalObjFields.next();
+            KeyValuePair kvPair = new KeyValuePair();
+            if (isSupportedOperation(literalObjEntry.getKey()))
+            {
+                throw new IllegalStateException("Literal Object value has key that is a operator key word " + literalObjEntry.getKey());
+            }
+            kvPair.key = literalObjEntry.getKey();
+            kvPair.value = getLiteralValueFromEntry(literalObjEntry).value;
+            kvPairs.add(kvPair);
+        }
+        return kvPairs;
+    }
+
+    private static boolean isSupportedOperation(String key)
+    {
+        return comparisonOperators.contains(key) || logicalOperators.contains(key);
+    }
+
+    @Override
+    public DatabaseCommand deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException
+    {
+
+        DatabaseCommand dbCommand = new DatabaseCommand();
+
+        // Check the first token
+//        if (jsonParser.currentToken() != JsonToken.START_OBJECT)
+//        {
+//            throw new IllegalStateException("Expected database command node to be an object");
+//        }
+        // while (jsonParser.currentToken() != JsonToken.END_OBJECT)
+        while (!jsonParser.isClosed())
```

Comment:
This will lead to bugs if your DB command is embedded inside a parent object.  Pretty much you assume you can consume the full JSON token stream, disregarding if there are token that belong to the parent object.

Reply:
Fix this